### PR TITLE
test(seagap): add test case to raise ValueError

### PR DESCRIPTION
**Pull request by lizimu0709**
Hi @lsetiawan ，
I've finished L1 task (Offshore Geodesy - Add test case to raise ValueError) and added two new test cases in tests/test_harmonic_mean.py. One test case error reported by the unit test has been fixed.

test_sv_harmonic_mean_invalid_input: This test case checks if the sv_harmonic_mean function raises a ValueError when provided with an invalid end_depth input (in this case, a string instead of a float or integer).

test__compute_hm_invalid_input: This test case checks if the _compute_hm function raises a ValueError when provided with an invalid end_depth input (in this case, a string instead of a float or integer).

These test cases will ensure that the ValueError scenarios in both sv_harmonic_mean and _compute_hm functions are properly covered during testing.

Reference: https://github.com/uw-ssec/codeuw/issues/23

### CODE REVIEW

**Comment by lsetiawan** [comment]
Thanks @lizimu0709 for this, this is a great start. However looks like the tests didn't raise the `ValueError`, but rather `TypeError`. Probably need to modify these. The expectations are:

`_compute_hm`: `ValueError` is raised when the expected columns is not in the columns within the data provided

https://github.com/uw-ssec/offshore-geodesy/blob/9d27d456c3fde1a6d41219173a130f921b8e2c25/src/gnatss/harmonic_mean.py#L34-L35

`sv_harmonic_mean`: `ValueError` is raised when the expected sound speed data is empty

https://github.com/uw-ssec/offshore-geodesy/blob/9d27d456c3fde1a6d41219173a130f921b8e2c25/src/gnatss/harmonic_mean.py#L72-L74

These 2 simple cases should be handled within the test.

**Comment by lizimu0709** [comment]
@lsetiawan Hi Don,
Thanks for your review and comments, that cleared my mind a lot!
But as for this test case:
> `sv_harmonic_mean`: `ValueError` is raised when the expected sound speed data is empty
> 
> https://github.com/uw-ssec/offshore-geodesy/blob/9d27d456c3fde1a6d41219173a130f921b8e2c25/src/gnatss/harmonic_mean.py#L72-L74

 I wrote the test case
```python
def test_sv_harmonic_mean_empty_dataframe():
    svdf = pd.DataFrame()
    start_depth = 0
    end_depth = 2
    with pytest.raises(ValueError):
        sv_harmonic_mean(svdf, start_depth, end_depth)
```
```
FAILED tests/test_harmonic_mean.py::test_sv_harmonic_mean_empty_dataframe - KeyError: 'speed'
```
But the test still fails, I presume it's because when testing sv_harmonic_mean, the function tries to access the column named 'speed' in the DataFrame, but this DataFrame is empty, so it doesn't have any columns. I would like to ask if the function should check if the DataFrame is empty before trying to access its columns. Since the requested test case expects a ValueError to be raised, but actually raises a KeyError. e.g. like this
```python
def sv_harmonic_mean(svdf, start_depth, end_depth):
    if svdf.empty:
        raise ValueError("Dataframe is empty! Please check your data inputs.")
    svdf = svdf[svdf[SP_SOUND_SPEED] > 0].reset_index(drop=True)
    # the rest of the function...
```

**Comment by lizimu0709** [comment]
If we don't want to modify the source code of the sv_harmonic_mean function, but instead want to raise the correct ValueError in the test, then I need to make sure that the test situation matches the situation in which the function might raise a ValueError. From the code snippet you provided, it looks like sv_harmonic_mean throws a ValueError when it checks that the length of abs_sv is 0. So, I could try creating a DataFrame for testing that has an empty abs_sv column. 
```python
def test_sv_harmonic_mean_empty_dataframe():
    # Create a DataFrame with empty 'abs_sv' column
    svdf = pd.DataFrame({"speed": [], "depth": []})
    start_depth = 0
    end_depth = 2
    with pytest.raises(ValueError):
        sv_harmonic_mean(svdf, start_depth, end_depth)
```

It turns out that this does work, may I ask if is this what we want for the test case?

File: tests/test_harmonic_mean.py [hunk 1]
```diff
@@ -56,3 +56,28 @@ def test__compute_hm(start_idx, end_idx, expected_hm):
...
+        _compute_hm(svdf, start_depth, end_depth)
+
+
+# Add test for empty dataframe
+# def test_sv_harmonic_mean_empty_dataframe():
+#     svdf = pd.DataFrame()
+#     start_depth = 0
+#     end_depth = 2
+#     with pytest.raises(ValueError):
+#         sv_harmonic_mean(svdf, start_depth, end_depth)
```
Comment (lsetiawan) [inline tests/test_harmonic_mean.py:76]: Could you remove the commented function? Thanks!

**Comment by lsetiawan** [comment]
> I would like to ask if the function should check if the DataFrame is empty before trying to access its columns.

Thanks for diving deep into this issue. I think what you discussed is definitely very reasonable to have. Currently this function indeed, doesn't check for an empty dataframe. Opening an issue about this, would be really great and much appreciated 😄 And then if you'd like to contribute further, I think adding this improvement and the test for it, would really increase the robustness of the software.

> It turns out that this does work, may I ask if is this what we want for the test case?

This is exactly what I was looking for! Thank you 🥳

**Review by lsetiawan — CHANGES_REQUESTED** [review]
Just need to clean up a little bit, see above. Thanks again!

File: tests/test_harmonic_mean.py (hunk 1 above)
Comment (lizimu0709) [reply tests/test_harmonic_mean.py]: Done! Thanks for the review! 
I will add the check in the sv_harmonic_mean if the DataFrame is empty before trying to access its columns and the corresponding test case in the next commit.
```python
def sv_harmonic_mean(svdf, start_depth, end_depth):
    if svdf.empty:
        raise ValueError("Dataframe is empty! Please check your data inputs.")
    svdf = svdf[svdf[SP_SOUND_SPEED] > 0].reset_index(drop=True)
    # the rest of the function...
```

**Review by lsetiawan — APPROVED** [review]
This is ready for merging. Thanks @lizimu0709!

File: tests/test_harmonic_mean.py (hunk 1 above)
Comment (lsetiawan) [reply tests/test_harmonic_mean.py]: Let's open a new Pull Request with the changes you mentioned above.
